documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -261,4 +263,302 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
 
 **Examples**: N/A
 
-Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
\ No newline at end of file
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2028" />R2028 NonEmptyClientName
+**Output Message**: Empty x-ms-client-name property.
+
+**Description**: The [`x-ms-client-name`](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-client-name) extension is used to change the name of a parameter or property in the generated code.
+
+**Why the rule is important**: This value cannot be empty, because we need to use it as the identifier for a property or model.
+
+**How to fix the violation**: Add a non-empty value for `x-ms-client-name`.
+
+**Impact on generated code**: Generated SDK code will expose the corresponding client name on property or model.
+
+**Examples**:
+```
+...
+"x-ms-client-name": "name"
+...
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2012" />R2012 XmsClientNameParameter
+**Output Message**: Value of 'x-ms-client-name' cannot be the same as '{0}' Property/Model.
+
+**Description**: The [`x-ms-client-name`](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-client-name) extension is used to change the name of a parameter or property in the generated code. By using the 'x-ms-client-name' extension, a name can be defined for use specifically in code generation, separately from the name on the wire. It can be used for query parameters and header parameters, as well as properties of schemas.
+
+**Why the rule is important**: This value cannot be same as parameter name or property name, because having the same name invalidates the usage.
+
+**How to fix the violation**: Please use different name that you'd like to see on the generated code.
```

Comment:
How about "Please specify non-empty `x-ms-client-name`, different from name of the model/property name that you'd like to be generated" 

I'd like to specify "different from mofrl/property name" because the message is like "you used the same name"